### ISSUE TITLE
Version numbers

### ISSUE THREAD

**Issue by attila-i-szabo** [PhilipVinc/mpi4jax#49]
I cannot install v0.2.10 (and a couple of other versions) using pip because I get the following error
```
WARNING: Discarding https://files.pythonhosted.org/packages/14/d7/59e58f912339c6f9a74857c22a46214790df41e29196e949f4bdae7045ab/mpi4jax-0.2.10.tar.gz#sha256=1356f3deda93635dacac43babd5f8f42937146ff33506376e5919ae10047edbb (from https://pypi.org/simple/mpi4jax/) (requires-python:>=3.6). Requested mpi4jax from https://files.pythonhosted.org/packages/14/d7/59e58f912339c6f9a74857c22a46214790df41e29196e949f4bdae7045ab/mpi4jax-0.2.10.tar.gz#sha256=1356f3deda93635dacac43babd5f8f42937146ff33506376e5919ae10047edbb has inconsistent version: filename has '0.2.10', but metadata has '0.0.0'
```
When I try installing it directly from the repo (`python3 -m pip install --upgrade git+https://github.com/PhilipVinc/mpi4jax.git@v0.2.10`), pip thinks it got version 0.0.0, so it is not accepted by anything that depends on it.

**Comment by PhilipVinc** [comment]
Hi Attila,

What platform are you on and what is the version of your python and pip?
I cannot reproduce locally on Fedora or MacOs, with python 3.8/3.9 and a freshly updated pip (python -m pip install -U pip).

@dionhaefner I think this is related to the automatic generation of the version number... do you have any idea of what is going on?

**Comment by PhilipVinc** [comment]
Related: [pip#7179](https://github.com/pypa/pip/issues/7179).

Upon inspecting our wheels on Pypi, it turns out that the [egg_info] tag_build field is declared but not populated in setup.cfg...

**Comment by PhilipVinc** [comment]
@attila-i-szabo 

pip install git+https://github.com/PhilipVinc/mpi4jax#0.2.10

should work, however.

**Comment by attila-i-szabo** [comment]
I'm on Ubuntu 18.04, python3.6, freshly upgraded pip (21.0.1). I don't have access to the pip command directly and I use a venv (institutional rules...). The command you sent does install v0.2.10, but calls it v0.0.0

**Comment by dionhaefner** [comment]
I'll have a look tomorrow, sounds like the metadata is still not propagated correctly in our wheels, which pip recently started to enforce.

**Comment by dionhaefner** [comment]
@attila-i-szabo which version of `setuptools` are you using?

**Comment by attila-i-szabo** [comment]
I use the latest setuptools (54.0.0). Turns out the installation works with my original command on python3.8 and the installed library gets the correct version number.

**Comment by dionhaefner** [comment]
Since neither of us has been able to reproduce this behavior, this is getting a bit tricky to debug. I've tried several old versions of `pip` and `setuptools`, on Python 3.6 and recent versions, but everything worked fine.

My suspicion is that you have some path confusion going on where you accidentally used an outdated pip or setuptools. Either way, I have hardened the version detection in #50 which should make this more robust in older setups.

It seems like the issue is resolved for now?
